spec/services/create_location_decorator_spec.rb

```diff
@@ -0,0 +1,54 @@
+RSpec.describe CreateLocationDecorator, '.build' do
+  let(:postcode) { 'BT7 3AP' }
+  let(:search_limit) { 3 }
+  let(:booking_location_id) { 222 }
+  let(:booking_location) { double(id: booking_location_id) }
+  let(:location_id) { 111 }
+  let(:location) { double(id: location_id, booking_location_id: booking_location_id) }
+  let(:nearest_locations) { double }
+  let(:search_context) { double }
+  let(:twilio_number) { double }
+  let(:location_decorator) { double }
+
+  def create_decorator
+    described_class.build(location: location,
+                          postcode: postcode,
+                          search_limit: search_limit)
+  end
+
+  subject(:decorator) { create_decorator }
+
+  before do
+    allow(Locations).to receive(:find).with(booking_location_id).and_return(booking_location)
+    allow(Locations).to receive(:nearest_to_postcode).with(postcode, limit: search_limit).and_return(nearest_locations)
+    allow(LocationSearchContext).to receive(:build).with(nearest_locations, location).and_return(search_context)
+    allow(Switchboard).to receive(:lookup).with(location_id).and_return(twilio_number)
+    expect(LocationDecorator).to receive(:new).with(
```

Comment:
Yeah, I was wanting to ensure the decorator was actually constructed when I was building this thing but the assertion in the `it { ... }` below would imply that.